datafusion/core/tests/sql/arrow_files.rs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+use datafusion::execution::options::ArrowReadOptions;
+
+use super::*;
+
+async fn register_arrow(ctx: &mut SessionContext) {
```

Comment:
Looks good -- I think we should also write a sqllogictest version of this (e.g. https://github.com/apache/arrow-datafusion/blob/b578c5819fc05801f2972dd427fbc13cf4773ea8/datafusion/core/tests/sqllogictests/test_files/ddl.slt#LL259C1-L260C1)

So that we have coverage of creating an arrow backed table via SQL

But I also think that could be added as a follow on PR

Reply:
registration support added in cfad814d21e51a2a3bbea9dacf6a9dc42977397a, and sqllogictest added in 718a21ffaaef71f71d693f0b6c9d705373c75702